src/readers/lsbi.c

```diff
@@ -0,0 +1,193 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <errno.h>
+#include <stdbool.h>
+
+#include <readers.h>
+#include <logs.h>
+
+#define PATTERN_SIZE 4
+#define EXTENSION_BLOCK_LENGTH 8
+#define LSB(x, n) ((x) & ((1 << (n)) - 1))
+
+
+Stego *retrieve_lsbi(FILE *file, size_t offset, char **extension)
+{
+    if (fseek(file, offset, SEEK_SET) < 0)
+    {
+        perror("Failed to seek to offset in file");
+        return NULL;
+    }
+
+    int pattern[PATTERN_SIZE] = {0};
+
+    uint8_t pattern_bytes[PATTERN_SIZE];
+    
+    if (fread(pattern_bytes, sizeof(uint8_t), PATTERN_SIZE, file) != PATTERN_SIZE)
+    {
+        perror("Not enough bytes to read pattern");
+        return NULL;
+    }
+
+    for(int i = 0 ; i < PATTERN_SIZE ; i++)
+    {
+        pattern[i] = pattern_bytes[i] & 1;
+        LOG("Pattern %d inversion: %d\n", i, pattern[i]);
+    }
+
+    /*
+                0  1  2  3  4  5 ...
+    B  G  R  B  G  R  B  G  R  B ...
+    00 01 10 11 S  -  I  Z  -  E ...
+    */
+
+    size_t bits_needed = sizeof(uint32_t) * 8;
+    size_t bytes_count = (bits_needed * 3 + 1) / 2;
+    uint8_t *length_bytes = malloc(bytes_count);
```

Comment:
Tal vez en vez de hacer malloc podríamos usar un array en stack, que C puede alocar en tiempo de compilación y no sería necesario free, especialmente con los valores constantes multiplicando el sizeof.

```suggestion
    uint8_t length_bytes[bytes_count];
```